lib/cppcheck.cpp

```diff
@@ -1417,6 +1419,125 @@ void CppCheck::executeAddons(const std::string& dumpFile, const FileWithDetails&
     }
 }
 
+void CppCheck::setClassification(ErrorMessage &errMsg) const {
```

Comment:
The ultimate goal would be that we only have functions to determine classification and guideline in one place. Right now we have it in both lib and gui.

Reply:
There is a use case in the GUI that the report type is changed after the analysis. Therefore the GUI needs to be able to take existing results and determine classification and guideline.
The GUI doesn't store the original `ErrorMessage` objects though as far as I remember. I therefore think it would be preferable if these functions did not use the whole ErrorMessage object.